src/components/PupilJourneyComponents/QuizMCQSingleAnswer/QuizMCQSingleAnswer.tsx

```diff
@@ -1,75 +1,106 @@
+import { useContext, useEffect, useState } from "react";
 import {
   OakBox,
   OakRadioButton,
   OakRadioGroup,
 } from "@oak-academy/oak-components";
 
 import { MCAnswer } from "@/node-lib/curriculum-api-2023/shared.schema";
+import { QuizEngineContext } from "@/components/PupilJourneyComponents/QuizEngineProvider";
 
 export type QuizMCQSingleAnswerProps = {
   questionUid: string;
   answers: MCAnswer[];
-  currentQuestionIndex: number;
-  isFeedbackMode: boolean;
-  selectedAnswer?: { answer?: MCAnswer | null; index: number };
-  setSelectedAnswer: (val: { answer?: MCAnswer | null; index: number }) => void;
 };
 
 export const QuizMCQSingleAnswer = (props: QuizMCQSingleAnswerProps) => {
-  const {
-    questionUid,
-    currentQuestionIndex,
-    answers,
-    isFeedbackMode,
-    selectedAnswer,
-    setSelectedAnswer,
-  } = props;
+  const { questionUid, answers } = props;
+
+  const quizEngineContext = useContext(QuizEngineContext);
+
+  const currentQuestionIndex = quizEngineContext?.currentQuestionIndex || 0;
+  const questionState = quizEngineContext?.questionState[currentQuestionIndex];
+
+  const [selectedAnswer, setSelectedAnswer] = useState<
+    MCAnswer | null | undefined
+  >(null);
+
+  useEffect(() => {
+    if (questionState?.mode === "grading") {
+      quizEngineContext?.handleSubmitMCAnswer(selectedAnswer);
```

Comment:
💣 triggering callbacks in `useEffect` can lead to confusing/unexepcted behaviour down the line when someone makes an innocuous change that causes a dependency to mutate. Is there anyway this could be triggered directly by a user interaction instead?. If this isn't destructive in anyway it might be ok to leave as is.

Reply:
The parent component triggers the state change in the provider.  Not sure how else I can trigger this . Any suggestions ?

Reply:
Inside the `onChange` on LN49 where you `setSelectedAnswer` maybe?

Possibly:

```typescript
switch (questionState.mode) {
case "init":
  quizEngineContext.updateQuestionMode("input");
  break;
case "grading":
  quizEngineContext.handleSubmitMCAnswer(selectedAnswer);
  break;
}
```

Reply:
That would require the user to change the selected item after clicking submit